events/cognito.go

```diff
@@ -125,11 +133,24 @@ type CognitoEventUserPoolsPreTokenGenRequest struct {
 	ClientMetadata     map[string]string  `json:"clientMetadata"`
 }
 
-// CognitoEventUserPoolsPreTokenGenResponse containst the response portion of  a PreTokenGen event
+// CognitoEventUserPoolsPreTokenGenV2Request contains request portion of V2 PreTokenGen event
+type CognitoEventUserPoolsPreTokenGenV2Request struct {
+	UserAttributes     map[string]string  `json:"userAttributes"`
+	GroupConfiguration GroupConfiguration `json:"groupConfiguration"`
```

Comment:
https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html#cognito-user-pools-lambda-trigger-syntax-pre-token-generation example says this is an array. 

`
...
        "groupConfiguration": [{
            "groupsToOverride": ["string", "string"],
            "iamRolesToOverride": ["string", "string"],
            "preferredRole": "string"
        }],
...
`

but the prose describes it as an object, and also states that it's the same as the v1 event, which in this library is also modeled as an object
> The input object that contains the current group configuration. The object includes groupsToOverride, iamRolesToOverride, and preferredRole.

🧐 Any chance you know if the docs are wrong on this one? A sample event from elsewhere in the docs that we could paste into the test cases too would be helpful

Reply:
Got a sample event from Brandon - turns out this one is a documentation issue.